### PR TITLE
Refactor mouseover

### PR DESCRIPTION
**Pull request by wkramer**

### CODE REVIEW

**Comment by wkramer** [comment]
TODO:
- [ ] Formatter for values
- [ ] Tooltip option to switch from center highlighting to `valueBefore` or `valueAfter`

File: src/Charts/chartBar.ts [hunk 1]
```diff
@@ -150,4 +161,43 @@ export class ChartBar extends Chart {
...
+        return window.getComputedStyle(element.node() as Element).getPropertyValue('fill')
+      })
+      .style('stroke', 'currentColor')
+  }
+
+  public onPointerOut() {
+    this.highlight.select('rect').style('opacity', 0)
+  }
+
+  public onPointerMove(x: number | Date, _xScale, yScale) {
```
Comment (hvangeffen) [inline src/Charts/chartBar.ts:181]: Is this highlight correct?:
image

File: src/Utils/setAlphaForColor.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+
+/**
+ * Sets the alpha value for a given color supporting alpha channels.
+ * @param color - The color to modify.
+ * @param alpha - The alpha value to set (between 0 and 1).
+ * @returns The modified color with the specified alpha value.
+ */
+export function setAlphaForColor(color, alpha) {
```
Comment (hvangeffen) [inline src/Utils/setAlphaForColor.ts:8]: Color and alpha could be typed

File: src/Utils/setAlphaForColor.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ */
+export function setAlphaForColor(color, alpha) {
+  if (color.startsWith('rgba') || color.startsWith('hsla')) {
+    return color.replace(/[\d\.]+\)/, `${alpha})`)
+  }
+  if (color.startsWith('#') && color.length === 9) {
+    return color.slice(0, -2) + Math.round(255 * alpha).toString(16).toUpperCase()
+  }
+  return color
+}
```
[the remark this answers is not in the document]
Comment (wkramer) [reply src/Utils/setAlphaForColor.ts:16]: Does not change a color with `alpha != 1` to to an color with `alpha = 1`.  
image

**Pull request merged**